cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -292,6 +293,12 @@ protected void startUp() throws Exception {
         Collections.sort(jarFiles);
         classpath.addAll(jarFiles);
         classpath.add("job.jar/classes");
+        // Add extra jars set in cConf
+        for (File extraJar : CConfigurationUtil.getExtraJars(cConf)) {
+          Location extraJarLocation = copyFileToLocation(extraJar, tempLocation, "");
```

Comment:
In `copyFileToLocation`, doesn't file.getName() in `Location targetLocation = targetDir.append(file.getName()).getTempFile(suffix);`  for a file "/tmp/foo.jar" return "foo.jar"?